### ISSUE TITLE
Prevent [additional] unverified emails from logging in?

### ISSUE THREAD

**Issue by ghost** [pennersr/django-allauth#650]
Firstly, I moved from django-registration to allauth, and so far its been very smooth. Thanks for all of the work you've done!

I'm using allauth purely for local registration, no social functionality.

I'd like to prevent unverified emails from being used to log in. I have set:
`ACCOUNT_EMAIL_VERIFICATION = 'mandatory'`

But that only seems to prevent an unverified email from logging in when first creating an account. I'd like the same functionality also when adding email addresses to an existing account (so newly added emails can't be used to log in until they're verified).

Is this expected behavior? I might be missing a setting somewhere.

**Comment by mcgeeco** [comment]
I'm confused. The docs seems to say that ACCOUNT_EMAIL_VERIFICATION = "mandatory" prevents a user from logging in until their email is verified. However, this conversation says that's not the case, which is in line with what's happening in my app. Which one is it?

**Comment by bnisevic** [comment]
I'm also confused. I use only email (no username) and this line is preventing confirmation to be sent:
https://github.com/pennersr/django-allauth/blob/master/allauth/account/utils.py#L132

If user is_active set to false, which is because we are awaiting his email confirmation, then email confirmation will not be sent because of this line 132. Very confusing.

My settings are:
```
ACCOUNT_SIGNUP_FORM_CLASS = 'apps.giivx_users.forms.signup.GiivxSignupForm'
ACCOUNT_USER_MODEL_USERNAME_FIELD = None
ACCOUNT_EMAIL_REQUIRED = True
ACCOUNT_USERNAME_REQUIRED = False
ACCOUNT_AUTHENTICATION_METHOD = 'email'
ACCOUNT_EMAIL_VERIFICATION = 'mandatory'
ACCOUNT_EMAIL_CONFIRMATION_EXPIRE_DAYS = 7
```

**Comment by pennersr** [comment]
See #2627 for a lengthy discussion on the topic. Closed via e97a666 -- note that this commit does not fix a security issue -- it merely results in a more consistent interpretation of mandatory email verification.

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/650
